runtime/dancebox/tests/common/xcm/core_buyer_common.rs

```diff
@@ -0,0 +1,367 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+use {
+    crate::{
+        assert_expected_events,
+        common::{
+            empty_genesis_data, run_to_session, set_dummy_boot_node, start_block,
+            xcm::{
+                mocknets::{
+                    DanceboxRococoPara as Dancebox, DanceboxSender, RococoRelay as Rococo,
+                    RococoRelayPallet, RococoSender,
+                },
+                *,
+            },
+        },
+    },
+    core::marker::PhantomData,
+    cumulus_primitives_core::Weight,
+    dancebox_runtime::{Registrar, RuntimeOrigin, ServicesPayment, XcmCoreBuyer},
+    frame_support::assert_ok,
+    nimbus_primitives::NimbusId,
+    pallet_xcm_core_buyer::RelayXcmWeightConfigInner,
+    parity_scale_codec::Encode,
+    polkadot_runtime_parachains::{
+        assigner_on_demand as parachains_assigner_on_demand, configuration,
+    },
+    sp_core::Pair,
+    sp_runtime::{traits::ValidateUnsigned, AccountId32},
+    staging_xcm::v3::QueryId,
+    staging_xcm_executor::traits::ConvertLocation,
+    tp_traits::{ParaId, SlotFrequency},
+    xcm_emulator::{Chain, RelayChain},
+};
+
+pub(crate) const PARATHREAD_ID: u32 = 3333;
```

Comment:
I tend to avoid `pub(crate)`, `pub` also compiles, no?

Reply:
yes, let me do that